bindings/sampling-heap-profiler.cc

```diff
@@ -74,20 +71,40 @@ NAN_METHOD(StopSamplingHeapProfiler) {
   info.GetIsolate()->GetHeapProfiler()->StopSamplingHeapProfiler();
 }
 
-NAN_METHOD(GetAllocationProfile) {
+void free_buffer_callback(char* data, void* buf) {
+  delete reinterpret_cast<std::vector<char>*>(buf);
+}
+
+NAN_METHOD(GetAllocationProfileProto) {
+  if (info.Length() != 2 || !info[0]->IsNumber() || !info[1]->IsNumber()) {
+    return Nan::ThrowTypeError(
+        "Expected exactly two arguments of type Integer.");
+  }
+  int64_t startTimeNanos = info[0].As<Integer>()->IntegerValue();
```

Comment:
The type check above if for `IsNumber` but you cast here to `Integer`. This isn't safe, as a value like 0.5 is a `Number` but not an `Integer`.

Reply:
I could use `IsInt32()`, but the start time in nanos is too large to be an int32 (even in ms, it's an int64). There does not seem to be an `IsInt64()`

Any recommendations?